mesonbuild/interpreter.py

```diff
@@ -2843,9 +2843,17 @@ def func_get_option(self, nodes, args, kwargs):
                                        'options of other subprojects.')
         opt = self.get_option_internal(optname)
         if isinstance(opt, coredata.UserFeatureOption):
-            return FeatureOptionHolder(self.environment, optname, opt)
+            opt = FeatureOptionHolder(self.environment, optname, opt)
         elif isinstance(opt, coredata.UserOption):
-            return opt.value
+            opt = opt.value
+        if optname == 'buildtype':
+            class BuildTypeString(str):
+                def startswith(self, token):
+                    if token == 'debug':
+                        mlog.warning("Use get_option('debug') to check whether debugging is enabled "
+                                     "instead of get_option('buildtype').startswith('debug')", location=node)
+                    super().startswith(token)
```

Comment:
I think if we are going to deprecate some usage of buildtype, better deprecate it completely. That's more consistent.